code/game/machinery/camera/camera_assembly.dm

```diff
@@ -20,6 +20,12 @@
 				4 = Screwdriver panel closed and is fully built (you cannot attach upgrades)
 	*/
 
+/obj/item/weapon/camera_assembly/Destroy()
+	for(var/thing in upgrades)
+		qdel(thing)
+	upgrades.Cut()

```

Comment:
We have multiple list qdels like this now, right? We should probably have a `QDEL_LIST` macro for this...